src/actor/npc/ac_npc_shop_master.c

```diff
@@ -0,0 +1,2677 @@
+            aNSC_setupAction(shop_master, game, next_act_idx[action]);
+        }
+    }
+}
+
+static void aNSC_sell_item_with_ticket(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *game) {
+    int val = mDemo_Get_OrderValue(0x4, 0x9);
+    mMsg_Window_c* msg_p = mMsg_Get_base_window_p();
+    if (val) {
+        if (mMsg_Check_MainNormalContinue(msg_p)) {
+            aNSC_set_last_day_str();
+            aNSC_Set_continue_msg_num(msg_p, shop_master, aNSC_get_msg_no((int)((f32)3.0 * fqrand()) + aNSC_MSG_TICKET_1));
+            mDemo_Set_OrderValue(mDemo_ORDER_NPC0, 0x9, 0x0);       
+            aNSC_setupAction(shop_master, game, 0xf);
+        }
+    }
+}
+
+static void aNSC_show_item_check(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *game) {
+    mMsg_Window_c* msg_p = mMsg_Get_base_window_p();
+    if (mMsg_Check_MainNormalContinue(msg_p)) {
+        mActor_name_t item = shop_master->sell_item;
+        int action = 0x1e;
+        int msg_no = aNSC_MSG_WAIT;
+        switch (mChoice_Get_ChoseNum(mChoice_Get_base_window_p())) {
+        case mChoice_CHOICE0:
+            switch (ITEM_NAME_GET_CAT(item)) {
+            case ITEM1_CAT_WALL:
+                CLIP(shop_indoor_clip)->change_wall_proc(item);
+                msg_no = aNSC_MSG_SHOW_CARPET_OFFER;
+                aNSC_request_show_camera(game, 0);
+                break;
+            case ITEM1_CAT_CARPET:
+                CLIP(shop_indoor_clip)->change_carpet_proc(item);
+                msg_no = aNSC_MSG_SHOW_CARPET_OFFER;
+                aNSC_request_show_camera(game, 0x1);
+                break;
+            default:
+                CLIP(shop_manekin_clip)->change2naked_manekin_proc(shop_master->ut_x, shop_master->ut_y);
+                action = 0x27;
+                msg_no = aNSC_MSG_CONFIRM_CLOTH;
+                shop_master->next_action = 0x1e;
+                shop_master->_9BC = Now_Private->cloth.item;
+                shop_master->_9BE = shop_master->sell_item;
+                break;
+            }
+            shop_master->_9B8 = TRUE;
+            break;
+        case mChoice_CHOICE1:
+            switch (ITEM_NAME_GET_CAT(item)) {
+            case ITEM1_CAT_CARPET:
+            case ITEM1_CAT_WALL:
+                msg_no = aNSC_MSG_SHOW_CARPET_DECLINE;
+                break;
+            default:
+                msg_no = aNSC_MSG_CANCEL_CLOTH;
+                break;
+            }
+            break;
+        case mChoice_CHOICE2:
+            action = 0x21;
+            msg_no = aNSC_MSG_BUY_CANCEL;
+            break;
+        }
+
+        if (msg_no != -1) {
+            aNSC_Set_continue_msg_num(msg_p, shop_master, aNSC_get_msg_no(msg_no));
+            mMsg_Set_ForceNext(msg_p);
+            aNSC_setupAction(shop_master, game, action); 
+        }
+    }
+}
+
+static void aNSC_chg_cloth_start_wait(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *game) {
+    if (mPlib_get_player_actor_main_index((GAME *)game) == mPlayer_INDEX_CHANGE_CLOTH) {
```

Comment:
This should be changed to GAME_PLAY* play and then you should change the cast to (GAME*)play.